### PR TITLE
is it correct?

### PR DESCRIPTION
**Pull request by Tage-official**

### CODE REVIEW

File: snake.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+};
+var body = {
+    value: '=',
+    x: 8,
+    y: 8
+};
+var food = {
+    x: Math.floor(Math.random() * (max - min) + min),
+    y: Math.floor(Math.random() * (max - min) + min)
+};
```
Comment (krakamod) [inline snake.js:16]: You shouldn't have `snake.js` file. Remove it. Only ts files should be used

File: snake.ts [hunk 2]
```diff
@@ -1 +1,69 @@
-console.log('Hello World, lul');
+type matrix = [[]]
+
+const max:number = 15
+const min:number = 0
```
Comment (krakamod) [inline snake.ts:4]: Make matrix size flexible

File: snake.ts [hunk 3]
```diff
@@ -1 +1,69 @@
-console.log('Hello World, lul');
+type matrix = [[]]
```
Comment (krakamod) [inline snake.ts:1]: The type means empty array. Remove it, you already have `Matrix` interface

File: snake.ts [hunk 4]
```diff
@@ -1 +1,69 @@
...
+
+const max:number = 15
+const min:number = 0
+
+interface Point{
```
Comment (krakamod) [inline snake.ts:6]: Add space between `Point` and `{`

File: snake.ts [hunk 5]
```diff
@@ -1 +1,69 @@
...
+
+}
```
Comment (krakamod) [inline snake.ts:8]: Define valid point type. It shouldn't be empty

File: snake.ts [hunk 6]
```diff
@@ -1 +1,69 @@
...
+
+interface Matrix{
```
Comment (krakamod) [inline snake.ts:10]: Space

File: snake.ts [hunk 7]
```diff
@@ -1 +1,69 @@
...
+    ground:matrix;
+}
```
Comment (krakamod) [inline snake.ts:12]: Matrix isn't ground. It is just array of arrays with some value as elements (e.g. `string`, or `number`, or even `Point`), decide what is better for you

File: snake.ts [hunk 8]
```diff
@@ -1 +1,69 @@
...
+
+interface Environment{
```
Comment (krakamod) [inline snake.ts:14]: Space

File: snake.ts [hunk 9]
```diff
@@ -1 +1,69 @@
...
+
+}
+
+interface EnvironmentField{
```
Comment (krakamod) [inline snake.ts:18]: Space

File: snake.ts [hunk 10]
```diff
@@ -1 +1,69 @@
...
+
+}
```
Comment (krakamod) [inline snake.ts:20]: Empty

**Review by krakamod — CHANGES_REQUESTED** [review]

**Review by krakamod — CHANGES_REQUESTED** [review]

File: snake.ts [hunk 11]
```diff
@@ -1 +1,19 @@
-console.log('Hello World, lul');
+import { ReadLine } from 'readline';
+const Readline: any = import('readline-sync')
```
Comment (krakamod) [inline snake.ts:2]: For now you don't need readline

File: snake.ts [hunk 12]
```diff
@@ -1 +1,19 @@
...
+
+interface Matrix {
+    matrix: [[]];
+    min: number;
+    max: number
+}
+
+interface Environment {
+    value: {}
```
Comment (krakamod) [inline snake.ts:11]: What does empty object type mean?

**Review by krakamod — CHANGES_REQUESTED** [review]

File: snake.ts (hunk 11 above)
Comment (Tage-official) [reply snake.ts:2]: genius

File: .gitignore [hunk 13]
```diff
@@ -0,0 +1 @@
+��
```
Comment (krakamod) [inline .gitignore:1]: Add node_modules to git ignore

File: snake.ts [hunk 14]
```diff
@@ -1 +1,15 @@
-console.log('Hello World, lul');
+interface Matrix {
+    matrix: string[][];
+    min: number;
+    max: number
+}
+
+interface Environment {
+    value: string[]
+}
```
Comment (krakamod) [inline snake.ts:9]: Set of env fields

**Review by krakamod — CHANGES_REQUESTED** [review]

File: snake.ts [hunk 15]
```diff
@@ -1 +1,20 @@
...
+
+interface Environment {
+    value: EnvironmentField[]
+}
+
+interface EnvironmentField {
+    value: '#';
+    x: number;
+    y: number
+}
```
Comment (krakamod) [inline snake.ts:15]: It should has a prop or extend Point interface

**Review by krakamod — CHANGES_REQUESTED** [review]

**Review by krakamod — APPROVED** [review]

**Pull request merged**